reconcile/sentry_config.py

```diff
@@ -0,0 +1,317 @@
+import logging
+import reconcile.queries as queries
+import utils.gql as gql
+import utils.secret_reader as secret_reader
+
+from reconcile.github_users import init_github
+from utils.config import get_config
+from utils.sentry_client import SentryClient
+
+
+SENTRY_PROJECTS_QUERY = """
+{
+  apps: apps_v1 {
+    sentryProjects {
+      team {
+        name
+      }
+      projects {
+        name
+        description
+        email_prefix
+        platform
+        sensitive_fields
+        safe_fields
+        auto_resolve_age
+        allowed_domains
+      }
+    }
+  }
+}
+"""
+
+SENTRY_USERS_QUERY = """
+{
+  roles: roles_v1 {
+    name
+    users {
+      name
+      github_username
+    }
+    bots {
+      name
+      github_username
+    }
+    permissions {
+      service
+      ...on PermissionSentryTeam_v1 {
+        team
+      }
+    }
+  }
+}
+"""
+
+SENTRY_TEAMS_QUERY = """
+{
+  teams: sentry_teams_v1 {
+    name
+  }
+}
+"""
+
+class SentryState:
+  def __init__(self):
+    # Map of user:teams[]
+    self.users = {}
+    # List of team names
+    self.teams = []
+    # Map of team:projects_config[]
+    self.projects = {}
+  
+  def init_users_from_current_state(self, users):
+    self.users = users
+
+  def init_users_from_desired_state(self, users):
+    # Input is in the form of team:members[]
+    for team in users.keys():
+      for user in users[team]:
+        if team not in self.users.keys():
+          self.users[user] = [team]
+        else:
+          self.users[user].append(team)
+
+  def init_projects_from_current_state(self, client, projects):
+    # Input is in the form of project:teams[]
+    for project in projects.keys():
+      for team in projects[project]:
+        # TODO: Retrieve project and store relevant config
+        p = client.get_project(project)
+        pdata = {
+          "name": p["name"],
+          "email_prefix": p["subjectPrefix"],
+          "platform": p["platform"]
+        }
+        if "sensitiveFields" in p.keys():
+          pdata["sensitive_fields"] = p["sensitiveFields"]
+        if "safeFields" in p.keys():
+          pdata["safe_fields"] = p["safeFields"]
+        if "resolveAge" in p.keys():
+          pdata["auto_resolve_age"] = p["resolveAge"]
+        if "allowedDomains" in p.keys():
+          pdata["allowed_domains"] = p["allowedDomains"]
+        if team not in self.projects.keys():
+          self.projects[team] = [pdata]
+        else:
+          self.projects[team].append(pdata)
+
+  def init_projects_from_desired_state(self, projects):
+    self.projects = projects
+
+  def init_teams(self, teams):
+    self.teams = teams
+
+class SentryReconciler:
+  def __init__(self, client, dry_run):
+    self.client = client
+    self.dry_run = dry_run
+
+  def reconcile(self, current, desired):
+    # Reconcile the teams frist
+    for team in current.teams:
+      if team not in desired.teams:
+        logging.info("deleting sentry team %s" % team)
```

Comment:
we generally log actions in the form of lists like this:

```python
logging.info(["delete_team", team])
```

That way we could parse it in the future if we ever want to.

This is a related comment of a review @maorfr did for @apahim:
https://github.com/app-sre/qontract-reconcile/pull/402#issuecomment-567616372

Reply:
Changed all logging to be in this format